rfc/rfc-80/rfc-80.md

```diff
@@ -0,0 +1,187 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-80: Support column families for wide tables
+
+## Proposers  
+- @xiarixiaoyao
+- @wombatu-kun
+
+## Approvers
+ - @vinothchandar
+ - @danny0405
+ - @beyond1920
+
+## Status
+
+JIRA: https://issues.apache.org/jira/browse/HUDI-7947
+
+## Abstract
+
+In streaming processing, there are often scenarios where the table is widened. The current mainstream real-time wide table concatenation is completed through Flink's multi-layer join;
+Flink's join will cache a large amount of data in the state backend. As the data set increases, the pressure on the Flink task state backend will gradually increase, and may even become unavailable.
+In multi-layer join scenarios, this problem is more obvious.  
+1.x also supports partial updates being encoded in logfiles. That should be able to handle this scenario. But even with partial-update, the column families will reduce write amplification on compaction.  
+
+So, main gains of clustering columns for wide tables are:  
+Write performance:
+- Writing is similar to ordinary bucket writing, but it involves splitting column clusters and sorting. Therefore, the writing performance of full data is 10% lower than that of native bucket writing.
+- However, if only some columns are updated among a large number of columns, the writing efficiency is much faster than that of non-column clustered tables. (Note: 1.0 introduces a partial update functionality, that can also avoid such writing costs, by writing only the changed columns)  
+
+Read performance:  
+Since the data is already sorted when it is written, the SortMerge method can be used directly to merge the data; compared with the native bucket data reading performance is improved a lot, and the memory consumption is reduced significantly.  
```

Comment:
I was wondering if we can introduce a pluggable merge sort to our compaction workflow, we can deem the regular table columns as in one family.

Reply:
good idea, but may be we can do it later as a follow-up in separate task?